runtime/src/main/jni/v8_inspector/src/inspector/v8-page-agent-impl.cpp

```diff
@@ -0,0 +1,257 @@
+    *out_content = resource.getContent(errorString);
+}
+
+void V8PageAgentImpl::searchInResource(ErrorString* errorString, const String& in_frameId,
+                                       const String& in_url,
+                                       const String& in_query,
+                                       const Maybe<bool>& in_caseSensitive,
+                                       const Maybe<bool>& in_isRegex,
+                                       const Maybe<String>& in_requestId,
+                                       std::unique_ptr<protocol::Array<protocol::GenericTypes::SearchMatch>>* out_result) {
+    bool isRegex = in_isRegex.fromMaybe(false);
+    bool isCaseSensitive = in_caseSensitive.fromMaybe(false);
+
+    std::map<std::string, v8_inspector::utils::PageResource> cachedPageResources = utils::PageResource::s_cachedPageResources;
+    if (utils::PageResource::s_cachedPageResources.size() == 0) {
+        cachedPageResources = utils::PageResource::getPageResources();
+    }
+
+    auto result = protocol::Array<protocol::GenericTypes::SearchMatch>::create();
+
+    auto it = cachedPageResources.find(in_url.utf8());
+    if (it == cachedPageResources.end()) {
+        *out_result = std::move(result);
+        return;
+    }
+
+    auto resource = it->second;
+    auto content = resource.getContent(errorString);
+    if (errorString->isEmpty()) {
+        auto matches = v8_inspector::utils::ResourceContentSearchUtils::searchInTextByLinesImpl(m_session, content, in_query, isCaseSensitive, isRegex);
+        for (auto& match : matches) {
+            result->addItem(std::move(match));
+        }
+
+        *out_result = std::move(result);
+    }
+}
+
+void V8PageAgentImpl::searchInResources(ErrorString*, const String& in_text,
+                                        const Maybe<bool>& in_caseSensitive,
+                                        const Maybe<bool>& in_isRegex,
+                                        std::unique_ptr<protocol::Array<protocol::Page::SearchResult>>* out_result) {
+    bool isRegex = in_isRegex.fromMaybe(false);
+    bool isCaseSensitive = in_caseSensitive.fromMaybe(false);
+
+    std::map<std::string, v8_inspector::utils::PageResource> cachedPageResources = utils::PageResource::s_cachedPageResources;
+    if (utils::PageResource::s_cachedPageResources.size() == 0) {
+        cachedPageResources = utils::PageResource::getPageResources();
+    }
+
+    std::vector<utils::PageResource> resourceObjects;
+
+    for (auto const& mapKVP : cachedPageResources) {
+        resourceObjects.push_back(mapKVP.second);
+    }
+
+    *out_result = utils::ResourceContentSearchUtils::getSearchMatches(m_session, resourceObjects, m_frameUrl.c_str(), in_text, isCaseSensitive, isRegex);
+}
+
+void V8PageAgentImpl::addScriptToEvaluateOnLoad(ErrorString*, const String& in_scriptSource,
+        String* out_identifier) {
+
+}
+
+void V8PageAgentImpl::removeScriptToEvaluateOnLoad(ErrorString*, const String& in_identifier) {
+
```

Comment:
The Dispatchers already have such functionality to inform the Front End that a method is not supported. I'll see if I can exclude the redundant methods from the custom protocol version, and generate the new interface. 

In case that's not possible we can assert that methods are not being called.